### PR TITLE
add azd devcontainer feature

### PR DESCRIPTION
**Pull request by hemarina**
- [x] Add it to web page
- [x] Add it to vscode extension
Related to https://github.com/Azure/azure-dev/issues/1095

### CODE REVIEW

File: _data/collection-index.yml [hunk 1]
```diff
@@ -473,3 +473,8 @@
   contact: https://github.com/tomharvey/devcontainer-features/issues
   repository: https://github.com/tomharvey/devcontainer-features
   ociReference: ghcr.io/tomharvey/devcontainer-features
+- name: Azure Developer CLI Dev Container Feature
+  maintainer: Azure Developer CLI Team
+  contact: https://github.com/Azure/azure-dev/issues
+  repository: https://github.com/Azure/azure-dev
+  ociReference: ghcr.io/Azure/azure-dev
```
Comment (samruddhikhandale) [inline _data/collection-index.yml]: Hi 👋

image

Curious, are the Feature(s) published? If so, can you help change the visibility of this metadata package (and your Feature package) to public? The [index](https://containers.dev/features) won't be able to crawl the Features in `Azure/azure-dev` if it's private.

More details - https://containers.dev/guide/author-a-feature#publishing

**Comment by savannahostrowski** [comment]
@bamurtaugh We will announce this on 7/12 when our July release ships!

File: _data/collection-index.yml (hunk 1 above)
Comment (samruddhikhandale) [inline _data/collection-index.yml]: Hi 👋

The ORAS spec requires the `ociReference` to be in lower case. Can we update here, thanks?
The supporting tools do help convert the case, however, its better to sync as much as possible. 

```suggestion
  ociReference: ghcr.io/azure/azure-dev
```

**Comment by hemarina** [comment]
@samruddhikhandale Our devcontainer feature is live. Would you mind taking another review?

**Review by samruddhikhandale — APPROVED** [review]
Looks great to me, thanks for adding it to the index ✨

**Pull request merged**